packages/core/src/modules/connections/handlers/DidExchangeCompleteHandler.ts

```diff
@@ -32,12 +33,14 @@ export class DidExchangeCompleteHandler implements MessageHandler {
     }
 
     const { message } = messageContext
-    if (!message.thread?.parentThreadId) {
+    const parentThreadId = message.thread?.parentThreadId
+    if (!parentThreadId) {
       throw new AriesFrameworkError(`Message does not contain pthid attribute`)
     }
     const outOfBandRecord = await this.outOfBandService.findByCreatedInvitationId(
       messageContext.agentContext,
-      message.thread?.parentThreadId
+      parentThreadId,
+      tryParseDid(parentThreadId) ? message.threadId : undefined
```

Comment:
Why do we now need to store the threadId? I'm trying to wrap my head around

Reply:
This is to relate a DidExchange Complete message to the flow where the OOBRecord comes from. Mainly because pthid will be the public did instead of an unique identifer.